meson.build

```diff
@@ -1005,9 +1016,14 @@ else
 
         # Debian installs this in /usr/sbin/ which is not in $PATH.
         # We check for 'bpftool' first, honouring $PATH, and in /usr/sbin/ for Debian.
-        bpftool = find_program('bpftool', '/usr/sbin/bpftool', required : bpf_framework_required)
+        # We use 'bpftool gen' subcommand, it was added by 985ead416df39d6fe8e89580cc1db6aa273e0175 (v5.6).
+        bpftool = find_program('bpftool',
+                               '/usr/sbin/bpftool',
+                               required : bpf_framework_required,
```

Comment:
Looks like `find_program` fails when `bpftool` exits with 1 on Ubuntu Focal even if `required` is set to `false` here. It's probably better than https://github.com/systemd/systemd/issues/22051 but I don't think it should happen if `-Dbpf-framework` wasn't set to `true` explicitly. There doesn't seem to be `check: false` that could be passed to `find_program` :-(

Reply:
Could you report that to meson team? It seems that we have no way to workaround the issue.

Reply:
I'm not sure whether it's a meson bug or not. It seems to me that it's reasonable to fail when `cmd --version` fails. I have no idea why that script is installed there in the first place and why `meson` should get it around. I mean, if `bpftool` isn't actually there that script shouldn't be there either.

Reply:
> It seems that we have no way to workaround the issue

As far as I understand it should be possible to call `find_program('bpftool')` and then pass the result to `run_command` to see if fails and pass it again to `find_program(..., version)` if it doesn't

Reply:
Ah, I understand the issue. ubuntu installs bpftool as a tiny wrapper... But IIUC, the error is triggered only when the matching real bpftool does not installed. I do not think we should not handle such issue here, as it is a build host issue.

Reply:
@yuwata let's wait for @wat-ze-hex. I think if there is a relatively easy way to somehow build a small bpf skeleton with `clang -target bpf`, `llvm-strip` and `bpftool gen` it would be better than looking for all the tools independently to just find out they can't work together.

Reply:
That would be similar to compiling and linking `int main() { return 0; }` to make sure the compiler and linker can work together.

Reply:
Yeah.